frontend/app/features/password-reset/components/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import Button from "@components/button/Button";
+import Input from "@components/input/Input";
+import { useUpdatePasswordMutation } from "@store/api/authSlice";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import styles from "../styles/resetPassword.module.css";
+
+const ResetPasswordPage = () => {
+  const searchParams = useSearchParams();
+  const token = searchParams?.get("token");
+  const router = useRouter();
+  const [newPassword, setNewPassword] = useState("");
+  const [message, setMessage] = useState("");
+
+  const [updatePassword, { isLoading, isSuccess, error }] = useUpdatePasswordMutation();
+
+ 
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const token = urlParams.get("token");
+    if (!token) {
+      router.replace("/404");
+    }
+  }, [router]);
```

Comment:
please move your useEffect below, keep your useEffect always at the bottom of your code